crates/driver/src/infra/config/file/mod.rs

```diff
@@ -122,6 +122,30 @@ enum Mempool {
     },
 }
 
+#[derive(Debug, Default, Deserialize)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields)]
+enum ManageNativeToken {
+    /// Wrap ETH address and keep inserting unwrap interactions
+    #[default]
+    WrapNativeToken,
+    /// Send native token address but keep inserting unwrap interactions
+    NativeToken,
```

Comment:
Does anybody actually want case 2 and 3 or is one of them a hypothetical use case? In this case I'd remove it. Otherwise, maybe having the config be more explicit, e.g. two booleans aka

struct NativeToken {
  bool wrap_address,
  bool insert_unwraps,
}

may be more self explanatory? I honestly find it hard to grasp the difference between these three types without deep context (imagine yourself reading `ManageNativeToken::NativeTokenEntirely` in the code).

Reply:
IMO it makes sense to have all the cases covered right away to finally get this out the door so we can make progress on other stuff. Deleting stuff when no solver is using this is still easy in a week or 2.
Splitting the 3 way enum into 2 bools makes sense to me. 👍